packages/rum-core/src/boot/startRum.ts

```diff
@@ -46,7 +46,9 @@ export function startRum(
     getCommonContext
   )
 
-  startLongTaskCollection(lifeCycle)
+  if (session.getPlan() === RumSessionPlan.REPLAY) {
```

Comment:
it could be interesting to have `session.hasReplayPlan()` / `session.hasLitePlan()` instead of letting callers do the check.
It seems that `getPlan` is also used in assembly but `isTracked` would be more appropriate here, right?

Reply:
> it could be interesting to have session.hasReplayPlan() / session.hasLitePlan() instead of letting callers do the check.

I don't really mind. I can add those methods if you want to, but then I'd like to introduce a helper to mock RumSession objects, because it's a bit inconvenient to explicitly every functions everywhere.

>  It seems that getPlan is also used in assembly but isTracked would be more appropriate here, right?

I mean, `getPlan` can return `undefined`, so I used it in the condition to avoid casting the result. But sure if we implement `hasReplayPlan` and `hasLitePlan`, we can just use `isTracked` instead.

Reply:
In assembly, the intent seems better expressed to me by checking `session.isTracked` rather than retrieving the plan and checking that it is undefined.

Then, since it seems that we don't need to manipulate the plan but rather explicitly check the current plan, the `hasXXXPlan` API seems nicer to me but not a big deal.